### DISCUSSION TITLE
Optional chaining On methods

### DISCUSSION THREAD

**Discussion by xiaoas** [expr-lang/expr#740] [General] [answered]
Currently, [Optional chaining](https://expr-lang.org/docs/language-definition#optional-chaining) on Object fields work well:
```js
MyObj?.Field1 ?? "default"
```
However, one cannot make an optional call to a method on an object:
```js
// this panics when MyObj is nil
MyObj?.Method1() ?? "default"
```
```js
// this does not compile
MyObj?.Method1?.() ?? "default"
```
So one is forced to write this, which works but is more verbose. Especially when `MyObj` is actually a long expression:
```js
MyObj != nil ? MyObj.Method1() : "default"
```
So may I petition to make `?.()` a valid optional function call syntax

**Comment by antonmedv** [comment, marked as answer]
Yes! It will be super nice to have it. Please create an issue.

**Comment by xiaoas** [comment]
I'll close this discussion. Future update should be in [this issue](https://github.com/expr-lang/expr/issues/741)
